SubmittingPatches-backports.rst

```diff
@@ -0,0 +1,344 @@
+wait, and provided you have sufficient permissions at https://tracker.ceph.com,
+you can open the Backport tracker issues and stage the backport yourself. In
+that case, read on.
+
+
+Creating Backport tracker issues
+--------------------------------
+
+Before we can start, the master tracker issue needs to exist and be in the right
+state. First, the master issue's Backport field should be populated. For example::
+
+    Backport: mimic, nautilus
+
+Second, the master PR should be in the "Pull request ID" field, e.g.::
+
+    Pull request ID: 99999
+
+Third, the master PR needs to have been merged already.
+
+Fourth, the master tracker issue's status should be set to "Pending Backport"::
+
+    Status: Pending Backport
+
+Once all of the above has been completed, issues can be created in the Backport
+tracker for tracking the backporting work. In our example, above, two issues
+would be created, to track the backports to "mimic" and "nautilus",
+respectively.
+
+You might be tempted to forge ahead and create those issues yourself. Please
+don't do that - it is difficult (bordering on impossible) to get it right when
+creating them manually, and why go that route when there is a script that gets
+it right every time? Setting up the script requires a small up-front time
+investment. Once that is done, creating backport issues becomes trivial.
+
+
+The backport-create-issue script
+--------------------------------
+
+The script used to create backport issues is located at
+``src/script/backport-create-issue`` in the master branch. Though there might be
+an older version of this script in a stable branch, do not use it. Only use the
+most recent version from master.
+
+Once you have the script somewhere in your PATH, you can proceed to install the
+dependencies.
+
+The dependencies are:
+
+* python3
+* python-redmine
+
+Python 3 should already be present on any recent Linux installation. The second
+dependency, python-redmine, can be obtained from PyPi::
+
+    https://pypi.org/project/python-redmine/
+
+Once you have the script in your PATH and the dependencies are installed, try to
+run the script::
+
+    backport-create-issue --help
+
+This should produce a usage message.
+
+Finally, run the script to actually create the Backport issues::
+
+    backport-create-issue --user ... --password ... 99999
+
+The script needs to know your https://tracker.ceph.com credentials in order to
+authenticate to Redmine. In lieu of providing your literal username and password
+on the command line, you could also obtain a REST API key ("My account" -> "API
+access key") and run the script like so::
+
+    backport-create-issue --key ... 99999
+
+
+Cherry-picking rules
+--------------------
+
+The following rules apply:
+
+* all bugfixes and features should land in master first
+* commits to stable branches should be cherry-picked from master
+* cherry-picks must be done using ``git cherry-pick -x``
```

Comment:
this line duplicated with the one on line 22
```rst
* cherry-picks must be done using ``git cherry-pick -x``
```

is this intended?

the same applies to the following lines.